### DISCUSSION TITLE
Spliting endopint /api/v1/status/openvpn into 3

### DISCUSSION THREAD

**Discussion by xinspl** [jaredhendrickson13/pfsense-api#366] [Ideas]
Hey I wonder if we split /api/v1/status/openvpn into 3:
  /api/v1/status/openvpn/server
  /api/v1/status/openvpn/p2p_servers
  /api/v1/status/openvpn/clients
then would it be possible to query that endpoint for server port and conn like :
`{"port": "11090", "conn__common_name": "username"}`

with only 2 servers and one conected user the response is 1KB but for example with 300 servers and 1000 conns the response will be very big(around 1MB). I only need to fast check if user is connected (no need to transfer so much data), and only periodicly gather all info.

**Comment by jaredhendrickson13** [comment]
Rather than creating separate endpoints for each, we could probably just add an optional parameter to the existing endpoint limit the scope of the endpoint to `server`, `p2p_server` and `clients`. That would allow you to target the specific data you're looking for, and query upon it as well. Feel free to open a [feature request](https://github.com/jaredhendrickson13/pfsense-api/issues/new?assignees=&labels=backlog%2C+feature+request&projects=&template=feature-request.md&title=). This is probably small enough it could be implemented in a patch.

Thanks!
